### DISCUSSION TITLE
index attribute for sentences

### DISCUSSION THREAD

**Discussion by jack-rory-staunton** [explosion/spaCy#4899] [Help: Coding & Implementations] [answered]
## Feature description
Although technically they are Spans (which do not have indices per se), it would be nice to somehow access the indices of a Doc's sentences: 

```
def get_sentence_index(ent):
    return [i for i, sent in enumerate(doc.sents)
                 if sent.start_char <= ent.start_char 
                 and sent.end_char >= ent.end_char][0]
```
perhaps by having the generator supply a tuple containing the index.

**Comment by svlandeg** [comment, marked as answer]
Sentences aren't internally stored as a list, so providing an index in the `doc` is not straightforward and implementing a function like this would not be efficient.

What you could do though, is loop through all the sentences in `list(doc.sents)` once, [set a custom extension](https://spacy.io/api/span#set_extension) to the corresponding `Span`, and use that in your other functions.

**Comment by jack-rory-staunton** [comment]
I'll do that - thanks!
